### PR TITLE
fix(replays): Initialize transaction from within the sub-process

### PR DESCRIPTION
**Pull request by cmanallen**
Previously we were using threads which required us to pass the hub around.  In a multi-processing universe we can't pickle the lock the hub acquires and pass it to the sub-process.  This is fine though because we can just use the hub that's available within the sub-process and not worry about passing it at all.

### CODE REVIEW

**Review by JoshFerge — APPROVED** [review]
curious if we could add a test to catch that regression in a follow up PR, or if its something we'd only see when deploying. but looks good!

**Comment by cmanallen** [comment]
@JoshFerge That's a really good point.  Some of this can't be tested with multi-processing but I did go ahead and expand the coverage to cover the synchronous case (effectively multi-processing) and the threaded case.

File: src/sentry/replays/consumers/recording.py [hunk 1]
```diff
@@ -102,9 +105,22 @@ def initialize_message_context(message: Message[KafkaPayload]) -> MessageContext
...
+    """Move the replay payload to permanent storage."""
+    transaction = sentry_sdk.start_transaction(
+        name="replays.consumer.process_recording",
+        op="replays.consumer",
+        sampled=random.random()
+        < getattr(settings, "SENTRY_REPLAY_RECORDINGS_CONSUMER_APM_SAMPLING", 0),
+    )
+    current_hub = sentry_sdk.Hub(sentry_sdk.Hub.current)
+    message_dict = RECORDINGS_CODEC.decode(message.payload.value)
+    ingest_recording(message_dict, transaction, current_hub)
```
Comment (bruno-garcia) [inline src/sentry/replays/consumers/recording.py:126]: is it worth having a span for decoding and another for `ingest_recording`? Or is this happening somehow automatically?

also noob here but how is `transaction` finishing? It does so without explicitly `transaction.finish()` nor `with` ?

Comment (bruno-garcia) [reply src/sentry/replays/consumers/recording.py:126]: Oh I see, it'll be closed within `ingest_recording`

**Review by bruno-garcia — APPROVED** [review]

File: src/sentry/replays/consumers/recording.py (hunk 1 above)
Comment (cmanallen) [reply src/sentry/replays/consumers/recording.py:126]: @bruno-garcia  yeah thats a fair point we should track the performance of this decoding step as well.

**Pull request merged**